File: storm-spring/src/test/java/st/orm/spring/repository/PetMetamodel.java
package st.orm.spring.repository;

import st.orm.Metamodel;
import st.orm.spring.model.Pet;

/**
 * Simulates a generated metamodel interface in the same package as repositories.
 * This should NOT be picked up by {@link st.orm.spring.RepositoryBeanFactoryPostProcessor}.
 */
public interface PetMetamodel extends Metamodel<Pet, Pet> {
}

File: storm-kotlin-spring/src/test/kotlin/st/orm/spring/repository/PetMetamodel.kt
package st.orm.spring.repository

import st.orm.Metamodel

/**
 * Simulates a generated metamodel interface in the same package as repositories.
 * This should NOT be picked up by [st.orm.spring.RepositoryBeanFactoryPostProcessor].
 *
 * Deliberately has no code references: it is exercised through classpath scanning, and
 * `RepositoryTest` asserts by bean name that no bean was registered for it.
 */
internal interface PetMetamodel : Metamodel<st.orm.spring.model.Pet, st.orm.spring.model.Pet>
